demo/paymentIntents/api.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+const {STRIPE_SECRET_KEY} = process.env;
+
+const serialize = (object: Object, scope: ?string): string => {
+  let result = [];
+  Object.keys(object).forEach((key) => {
+    const value = object[key];
+    const scopedKey = scope ? `${scope}[${key}]` : key;
+    if (value && typeof value === 'object') {
+      const nestedResult = serialize(value, scopedKey);
+      if (nestedResult !== '') {
+        result = [...result, nestedResult];
+      }
+    } else if (value !== undefined && value !== null) {
+      result = [...result, `${scopedKey}=${encodeURIComponent(String(value))}`];
+    }
+  });
+  return result.join('&').replace(/%20/g, '+');
+};
+
+const createPaymentIntent = (options: {}): Promise<string> => {
+  if (!STRIPE_SECRET_KEY) {
+    return Promise.reject(
+      new Error(
+        `A secret key is required to create PaymentIntents for handleCardPayment. Please set the following environment variable: \n\nSTRIPE_SECRET_KEY=<your secret key>`
+      )
+    );
+  }
+
+  // WARNING
+  // DO NOT USE THE FOLLOWING CODE IN A PRODUCTION APPLICATION!
+  // Your secret key should NEVER be used in on your frontend.
+  // We are doing this here purely for demonstration reasons.
+  // In the real world, creating PaymentIntents needs to be done
+  // on your backend server.
```

Comment:
We might want to make this even stronger and make clear the consequences: having your secret key on the frontend enables people to make charges on your behalf, which will be used by card testers, which will get your business banned from accepting credit card payments.